site/components/src/main/java/uk/nhs/hee/web/components/BulletinComponent.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    private Pageable<HippoBean> getDocumentBeans(final HstRequest request, final BulletinComponentInfo paramInfo) {
+        try {
+            return executeQuery(request, paramInfo);
+        } catch (QueryException qe) {
+            log.error("An error has occurred while trying to execute hst query.", qe);
+            return null;
+        }
+    }
+
+    protected Pageable<HippoBean> executeQuery(final HstRequest request, final BulletinComponentInfo paramInfo) throws QueryException {
+        final HstQuery query = buildQuery(request, paramInfo);
+
+        final HstQueryResult execute = query.execute();
+        return getPageableFactory().createPageable(
+                execute.getHippoBeans(),
+                execute.getTotalSize(),
+                paramInfo.getPageSize(),
+                getCurrentPage(request));
+    }
+
+    private HstQuery buildQuery(final HstRequest request, final BulletinComponentInfo paramInfo) {
+        final String documentPath = paramInfo.getPath();
+        final HippoBean scopeBean = doGetScopeBean(documentPath);
+
+        final String[] documentTypes = getDocumentTypes(paramInfo);
+        final HstQuery query = createQuery(scopeBean, documentTypes);
+
+        final int pageSize = paramInfo.getPageSize();
+        final int page = getCurrentPage(request);
+        query.setLimit(pageSize);
+        query.setOffset((page - 1) * pageSize);
+
+        query.setFilter(createQueryFilters(request, query));
+        applySortOrdering(request, query);
+
+        return query;
+    }
+
+    private String[] getDocumentTypes(BulletinComponentInfo paramInfo) {
+        final String documentTypes = paramInfo.getDocumentTypes();
+        return SiteUtils.parseCommaSeparatedValue(documentTypes);
+    }
+
+    private HstQuery createQuery(HippoBean scope, String[] documentTypes) {
+        HstQueryBuilder builder = HstQueryBuilder.create(scope);
+        return builder.ofTypes(documentTypes).build();
+    }
+
+    protected int getCurrentPage(final HstRequest request) {
+        return getAnyIntParameter(request, REQUEST_PARAM_PAGE, 1);
+    }
+
+    protected Filter createQueryFilters(final HstRequest request, final HstQuery query) {
+        Filter baseFilter = query.createFilter();
+        baseFilter.addOrFilter(getCategoryFilter(request, query));
+
+        return baseFilter;
+    }
+
+    private Filter getCategoryFilter(final HstRequest request, final HstQuery query) {
+        List<String> categoriesFilter = getQueryParameterValues(request, CATEGORY_QUERY_PARAM);
+        return createOrFilter(query, categoriesFilter, CATEGORY_BEAN_ATTRIBUTE);
+    }
+
+    private List<String> getQueryParameterValues(final HstRequest request, final String parameter) {
+        String[] parameterValues = request.getParameterValues(parameter);
+        if (parameterValues == null) {
+            return Collections.emptyList();
+        }
+
+        return Arrays.asList(parameterValues);
+    }
```

Comment:
Should we extract this logic to a utilities method under HstUtils class for example ? `getParameterValuesAsList`